### PR TITLE
Suffix experimental version / fix warning output

### LINKED ISSUE #13385: Experimental channel build binary should mention experimental in docker -v
With the introduction of a new "Experimental" channel in https://github.com/docker/docker/issues/13023, there should be a way for the user to determine that they are running an experimental build.

```docker -v```
today
```Docker version 1.6.0, build 4749651```
or
```Docker version 1.6.2-cs4, build 47224f0```
For experimental
```Docker version 1.6.2-exp, build 47224f0```
or
```Docker version 1.6.0, experimental build 4749651```

### PR DESCRIPTION
**Pull request by icecrime**
Closes #13385.
Relates to #13023.

### CODE REVIEW

**Comment by icecrime** [comment]
image

**Comment by duglin** [comment]
testcase?

**Comment by duglin** [comment]
Hmm, when I run I see:
```
WARN[0000] Running experimental build                   
```
Won't this bust some testcases?  Running 'em now....   I would hope at least one testcase actually checks the output from the CLI very carefully to notice this  :-)

**Comment by jfrazelle** [comment]
a bash test?

**Comment by duglin** [comment]
ok good - I am seeing some failures. Can we remove that warning? It'll make it really hard to know if a new features breaks something if I can run the testcases.

**Comment by icecrime** [comment]
@duglin I can remove that warning, to be honest I'm even surprised it goes to stderr, I was expecting it'll appear in logs only?

**Comment by icecrime** [comment]
My bad, this was logged on the client too, and the client doesn't have a log file. Move the warning to daemon-side only.

**Comment by duglin** [comment]
re: testing - I was thinking that we could have a testcase that runs "docker --version", "docker info" and "docker version" and based on the `ExperimentalBuild()` value it would check the output and make sure the word 'experimental' was there, or not.

**Comment by duglin** [comment]
@icecrime can you squash and add a test(s) ?

**Comment by icecrime** [comment]
@duglin Updated! Thanks

File: integration-cli/docker_cli_experimental_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+package main
+
+import (
+	"os/exec"
+	"strings"
+
+	"github.com/go-check/check"
+)
+
+func (s *DockerSuite) TestExperimentalVersion(c *check.C) {
```
Comment (duglin) [inline integration-cli/docker_cli_experimental_test.go]: I think you may need an additional test to make sure that "-experimental" **isn't** printed when the build flag isn't set.

Comment (icecrime) [inline integration-cli/docker_cli_experimental_test.go]: The thing is I *could* add `-experimental` in my `VERSION` file and get it anyway.

Comment (duglin) [inline integration-cli/docker_cli_experimental_test.go]: that's just crazy talk :-)

File: integration-cli/docker_cli_info_test.go [hunk 2]
```diff
@@ -26,12 +28,13 @@ func (s *DockerSuite) TestInfoEnsureSucceeds(c *check.C) {
 		"CPUs:",
 		"Total Memory:",
 		"Kernel Version:",
-		"Storage Driver:"}
+		"Storage Driver:",
+		fmt.Sprintf("Experimental: %t", utils.ExperimentalBuild()),
```
Comment (tiborvass) [inline integration-cli/docker_cli_info_test.go]: why not add Experimental conditionally instead of show Experimental: no ?

Comment (icecrime) [inline integration-cli/docker_cli_info_test.go]: That was merged as part of another PR, and no particular reason :-/

Comment (tiborvass) [inline integration-cli/docker_cli_info_test.go]: hmm ok, minor detail, but would prefer not have the word "experimental" in a "stable" release

Comment (calavera) [inline integration-cli/docker_cli_info_test.go]: I think I agree with Tibor, I'd rather not show any message.

**Comment by calavera** [comment]
This LGTM, but I'll wait to merge it because I don't see any other LGTM :smile: 

/cc @tiborvass, @jfrazelle

**Comment by duglin** [comment]
LGTM there ya go

File: hack/make.sh [hunk 3]
```diff
@@ -96,6 +96,7 @@ fi
 if [ "$DOCKER_EXPERIMENTAL" ]; then
 	echo >&2 '# WARNING! DOCKER_EXPERIMENTAL is set: building experimental features'
 	echo >&2
+	VERSION+="-experimental"
```
Comment (tianon) [inline hack/make.sh]: Ah, here's where it snuck in, @jfrazelle.  I think we should probably just revert this one line, IMO.

**Pull request merged**